### PR TITLE
Added r_cons_more and made some functions from less.c global (fix #11552)

### LINKED ISSUE #11552: Implement r_cons_more()
Like r_cons_less, but without going back in time

### PR DESCRIPTION
**Pull request by bart1e**

### CODE REVIEW

File: libr/include/r_cons.h [hunk 1]
```diff
@@ -736,6 +739,9 @@ R_API void r_cons_flush(void);
 R_API void r_cons_last(void);
 R_API int r_cons_less_str(const char *str, const char *exitkeys);
 R_API void r_cons_less(void);
+R_API int r_cons_next_match(int from, RList **mla, int lcount);
```
Comment (radare) [inline libr/include/r_cons.h]: use R_IPI, those functions should not be exported. they  are just interenal to rcons

Comment (radare) [reply libr/include/r_cons.h]: and  use different naming, because r_cons implies public

Comment (bart1e) [reply libr/include/r_cons.h]: OK, where should I declare these functions then?

Comment (radare) [reply libr/include/r_cons.h]: just change this to R_IPI for now, but we can add a private.h inside libr/cons with those definitions and include "./private.h" where needed.

File: libr/cons/cons.c [hunk 2]
```diff
@@ -1432,6 +1498,107 @@ R_API char *r_cons_lastline_utf8_ansi_len(int *len) {
...
 }
 
+R_API int r_cons_next_match(int from, RList **mla, int lcount) {
+	int l;
+	if (from > lcount - 2) {
+		return from;
+	}
+	for (l = from + 1; l < lcount; l++){
+		/* if there's at least one match on the line */
+		if (r_list_first(mla[l])) {
```
Comment (radare) [inline libr/cons/cons.c]: missing space before (

File: libr/cons/cons.c [hunk 3]
```diff
@@ -1432,6 +1498,107 @@ R_API char *r_cons_lastline_utf8_ansi_len(int *len) {
...
+	return from;
+}
+
+R_API int r_cons_prev_match(int from, RList **mla) {
+	int l;
+	if (from < 1) {
+		return from;
+	}
+	for (l = from - 1; l > 0; l--) {
+		if (r_list_first(mla[l])) {
```
Comment (radare) [inline libr/cons/cons.c]: missing space

File: libr/cons/cons.c [hunk 4]
```diff
@@ -1432,6 +1498,107 @@ R_API char *r_cons_lastline_utf8_ansi_len(int *len) {
...
+	lines[row++] = 0;
+	for (i = 0; s[i]; i++) {
+		if (row >= lines_size) {
+			int *tmp;
+			lines_size += 128;
+			if (lines_size * sizeof(int) < lines_size) {
+				free (lines);
+				return NULL;
+			}
+			tmp = realloc (lines, lines_size * sizeof(int));
```
Comment (radare) [inline libr/cons/cons.c]: missing space

File: libr/cons/less.c [hunk 5]
```diff
@@ -289,27 +122,27 @@ R_API int r_cons_less_str(const char *str, const char *exitkeys) {
 				}
 				rx = r_regex_new(sreg, "");
 			} else { /* we got an empty string */
-				from = next_match(from, mla, lines_count);
+				from = r_cons_next_match(from, mla, lines_count);
```
Comment (radare) [inline libr/cons/less.c]: missing space

File: libr/cons/less.c [hunk 6]
```diff
@@ -289,27 +122,27 @@ R_API int r_cons_less_str(const char *str, const char *exitkeys) {
...
 			if (rx) {
-				from = next_match (from, mla, lines_count);
+				from = r_cons_next_match (from, mla, lines_count);
 			}
 			break;
 		case 'N':
 		case 'p': 	/* previous match */
 			if (rx) {
-				from = prev_match(from, mla);
+				from = r_cons_prev_match(from, mla);
```
Comment (radare) [inline libr/cons/less.c]: missing space

**Comment by bart1e** [comment]
Thanks for review. I'll commit changes tomorrow.

**Comment by radare** [comment]
Thanks!

Btw how do you plan to make it available via ~?

~.. is less
~... is hud

Maybe we shoukd use this for more

~.... 

But its a bit unexpected.. also maybe good to integrate with the scr.pager

**Comment by ret2libc** [comment]
For the various style fixes, please just use `./sys/clang-format-diff.py`. You can run it with `git diff origin/master.. | ./sys/clang-format-diff.py -p1 -i` and it will fix the style for your changes.

**Comment by radare** [comment]
You can also indent the whole file with r2i.

**Comment by radare** [comment]
Imho indenting files that are modified but only indenting the lines that are touched would be a good option to be added to r2i

**Comment by bart1e** [comment]
Oops, didn't want to merge these all commits. Sorry for that...

**Comment by radare** [comment]
Never use merge

**Comment by radare** [comment]
please fix this PR, its all messed up, just cherry-pick the commits in a cleanly rebased branch and push -f

**Comment by radare** [comment]
just add more.c pager.c in the libr/cons/meson.build and everything should be green i think

**Pull request merged**